backend/tests/grynos.test.js

```diff
@@ -0,0 +1,37 @@
+const loadMockCoursesToDatabase = require('../src/seed/db-seed')
+    .loadMockCoursesToDatabase;
+const db = require('../src/db');
+const datefns = require('date-fns');
+
+describe('getCourses API call', () => {
+    beforeEach(() => {
+        return loadMockCoursesToDatabase();
+    });
+
+    test('should load course data', async () => {
+        const courses = await db.courses.getCourses('2018-04-18', '2018-04-19');
+        expect(courses).not.toBeNull();
+        expect(courses).toHaveLength(4);
+        expect(courses[0].name).toEqual('English Conversation');
+        expect(courses[0].price).toEqual(43);
+        expect(courses[0].location).toHaveLength(1);
+        expect(courses[0].teachingSession).toHaveLength(1);
+    });
+
+    test('validate the map reduce fucntion', async () => {
```

Comment:
This test case shouldn't be `async` function as it is unit testing a synchronous function. To make it so, the following changes are required:
1. Remove `async` from line 21 and `await` from line 23 (`reduceCoursesByDate` is not a `async` function)
2. Remove call for `getCourses` from this test case and instead use static array of courses (either from mock data JSON or then create few courses manually here by declaring a simple array of objects)